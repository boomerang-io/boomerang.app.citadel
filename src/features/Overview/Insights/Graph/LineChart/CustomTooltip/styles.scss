@import "~Styles/variables";

.c-tooltip {
  display: flex;
  flex-direction: column;
  background-color: $dark-gray;
  width: auto;
  height: auto;
  color: $white;
  border: solid $white 1px;
  border-radius: 0.325rem;
}
.b-tooltip--data {
  text-align: start;
  margin: 1rem;
  word-wrap: break-word;
}
.b-tooltip--field {
  font-weight: bold;
}
