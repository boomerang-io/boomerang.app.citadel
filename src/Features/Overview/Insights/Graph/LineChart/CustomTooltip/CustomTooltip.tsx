import React, { Component } from "react";
import "./styles.scss";

type Props = {
    payload?: any[];
};

class CustomTooltip extends Component<Props> {

  renderTooltip = (tooltipFields: any) => {
    return tooltipFields.map((field: any, index: any) => {
      return (
        // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
        <p key={`${field}-${index}`} className="b-tooltip--data">
          {" "}
          {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
          <span className="b-tooltip--field">{`${
            field === "passRate" ? "pass rate" : field === "criticalViolations" ? "critical violations" : field
          }: `}</span>
          {/* @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'. */}
          {`${this.props.payload[0].payload[field]}`}
        </p>
      );
    });
  };
  render() {
    // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
    const tooltipFields = this.props.payload.length > 0 ? Object.keys(this.props.payload[0].payload) : null;
    return (
      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
      <div className="c-tooltip">
        {/* @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'. */}
        {this.props.payload.length > 0 ? <div>{this.renderTooltip(tooltipFields)}</div> : null}
      </div>
    );
  }
}

export default CustomTooltip;
