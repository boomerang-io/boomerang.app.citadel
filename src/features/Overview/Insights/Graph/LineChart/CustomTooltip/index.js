import React, { Component } from "react";
import PropTypes from "prop-types";
// import moment from "moment";
// import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import "./styles.scss";

class CustomTooltip extends Component {
  renderTooltip = tooltipFields => {
    return tooltipFields.map((field, index) => {
      return (
        <p key={`${field}-${index}`} className="b-tooltip--data">
          {" "}
          <span className="b-tooltip--field">{`${
            field === "passRate" ? "pass rate" : field === "criticalViolations" ? "critical violations" : field
          }: `}</span>
          {`${this.props.payload[0].payload[field]}`}
        </p>
      );
    });
  };
  render() {
    const tooltipFields = this.props.payload.length > 0 ? Object.keys(this.props.payload[0].payload) : null;
    return (
      <div className="c-tooltip">
        {this.props.payload.length > 0 ? (
          <div>{this.renderTooltip(tooltipFields)}</div>
        ) : null}
      </div>
    );
  }
}

CustomTooltip.propTypes = {
  payload: PropTypes.array
};

export default CustomTooltip;
