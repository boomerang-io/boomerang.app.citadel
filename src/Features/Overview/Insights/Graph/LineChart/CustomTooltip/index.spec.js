import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import CustomTooltip from "./index";

const payload = [
  {
    color: "#047cc0",
    dataKey: "total",
    fill: "url(#blueGradient)",
    fillOpacity: 0.6,
    formatter: undefined,
    name: "total",
    payload: { date: 1542160248412, failed: 5, success: 12, total: 17 },
    stroke: "#047cc0",
    unit: undefined,
    value: 17
  }
];

describe("CustomTooltip --- Snapshot", () => {
  it("Capturing Snapshot of CustomTooltip", () => {
    const renderedValue = renderer.create(
      <MemoryRouter>
        <CustomTooltip payload={payload} />
      </MemoryRouter>
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("CustomTooltip --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <CustomTooltip payload={payload} />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
