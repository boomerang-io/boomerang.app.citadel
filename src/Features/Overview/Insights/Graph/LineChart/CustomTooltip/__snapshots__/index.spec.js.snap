// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip --- Snapshot Capturing Snapshot of CustomTooltip 1`] = `
<div
  className="c-tooltip"
>
  <div>
    <p
      className="b-tooltip--data"
    >
       
      <span
        className="b-tooltip--field"
      >
        date: 
      </span>
      1542160248412
    </p>
    <p
      className="b-tooltip--data"
    >
       
      <span
        className="b-tooltip--field"
      >
        failed: 
      </span>
      5
    </p>
    <p
      className="b-tooltip--data"
    >
       
      <span
        className="b-tooltip--field"
      >
        success: 
      </span>
      12
    </p>
    <p
      className="b-tooltip--data"
    >
       
      <span
        className="b-tooltip--field"
      >
        total: 
      </span>
      17
    </p>
  </div>
</div>
`;
